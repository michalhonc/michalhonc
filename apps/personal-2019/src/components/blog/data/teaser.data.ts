export const Query = {
  allMarkdownRemark: {
    edges: [
      {
        node: {
          frontmatter: {
            date: "2020-03-31",
            language: "en",
            description:
              "Canny tricks that won't make your React app better but will make your dev process easier.",
            path: "/blog/5-canny-tricks-for-react-devs",
            title: "5 Canny little tricks for React devs",
            category: "react",
            thumbnail: {
              childImageSharp: {
                fluid: {
                  src: "/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png",
                  aspectRatio: 1.4184397163120568,
                  srcSet:
                    "/static/4e57a90ee8c2f36cfe0eba16def56413/69585/5_canny_react_tricks.png 200w,\n/static/4e57a90ee8c2f36cfe0eba16def56413/497c6/5_canny_react_tricks.png 400w,\n/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png 704w",
                  sizes: "(max-width: 704px) 100vw, 704px",
                },
              },
            },
          },
        },
      },
      {
        node: {
          frontmatter: {
            date: "2020-03-31",
            language: "cs",
            description:
              "Triky, které Vaši aplikaci nezlepší, ale usnadní vám proces vývoje",
            path: "/blog/5-triku-pro-vyvojare-reactu",
            title: "5 triků pro vývojáře Reactu",
            category: "react",
            thumbnail: {
              childImageSharp: {
                fluid: {
                  src: "/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png",
                  aspectRatio: 1.4184397163120568,
                  srcSet:
                    "/static/4e57a90ee8c2f36cfe0eba16def56413/69585/5_canny_react_tricks.png 200w,\n/static/4e57a90ee8c2f36cfe0eba16def56413/497c6/5_canny_react_tricks.png 400w,\n/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png 704w",
                  sizes: "(max-width: 704px) 100vw, 704px",
                },
              },
            },
          },
        },
      },
      {
        node: {
          frontmatter: {
            date: "2020-02-25",
            language: "cs",
            description:
              "Nepřicházejte kvůli AdBlocku o peníze z reklamy! Zjištěte 7 způsobů jak se proti AdBlocku můžete bránit.",
            path: "/blog/7-zpusobu-jak-obejit-adblock",
            title: "7 způsobů jak obejít AdBlock",
            category: "reklama",
            thumbnail: {
              childImageSharp: {
                fluid: {
                  src: "/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png",
                  aspectRatio: 1.492537313432836,
                  srcSet:
                    "/static/d96fb15f308d10e6d5035097232fbd35/69585/adblock_thumbnail.png 200w,\n/static/d96fb15f308d10e6d5035097232fbd35/497c6/adblock_thumbnail.png 400w,\n/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png 500w",
                  sizes: "(max-width: 500px) 100vw, 500px",
                },
              },
            },
          },
        },
      },
      {
        node: {
          frontmatter: {
            date: "2020-02-25",
            language: "en",
            description:
              "Don't lose advertising money because of AdBlock! Find out 7 ways you can defend yourself against AdBlock.",
            path: "/blog/7-ways-to-bypass-adblock",
            title: "7 ways to bypass AdBlock",
            category: "ads",
            thumbnail: {
              childImageSharp: {
                fluid: {
                  src: "/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png",
                  aspectRatio: 1.492537313432836,
                  srcSet:
                    "/static/d96fb15f308d10e6d5035097232fbd35/69585/adblock_thumbnail.png 200w,\n/static/d96fb15f308d10e6d5035097232fbd35/497c6/adblock_thumbnail.png 400w,\n/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png 500w",
                  sizes: "(max-width: 500px) 100vw, 500px",
                },
              },
            },
          },
        },
      },
      {
        node: {
          frontmatter: {
            date: "2020-04-08",
            language: "en",
            description:
              "In this tutorial I'll show you how to set up professional development process with help of continuous deployment.",
            path: "/blog/continuous-deployment-for-free",
            title:
              "Continuous development for FREE with Github, Travis CI and Heroku",
            category: "continuous deployment",
            thumbnail: {
              childImageSharp: {
                fluid: {
                  src: "/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png",
                  aspectRatio: 1.3605442176870748,
                  srcSet:
                    "/static/f53b530a16d87556855e6e551c169f83/69585/travis_thumbnail.png 200w,\n/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png 387w",
                  sizes: "(max-width: 387px) 100vw, 387px",
                },
              },
            },
          },
        },
      },
      {
        node: {
          frontmatter: {
            date: "2020-04-08",
            language: "cs",
            description:
              "V tomto tutorialu vám ukáži, jak nastavit profesionální vývojové prostřední s pomocí continuous deployment.",
            path: "/blog/continuous-deployment-zdarma",
            title: "Continuous development zdarma s Github, Travis CI a Heroku",
            category: "continuous deployment",
            thumbnail: {
              childImageSharp: {
                fluid: {
                  src: "/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png",
                  aspectRatio: 1.3605442176870748,
                  srcSet:
                    "/static/f53b530a16d87556855e6e551c169f83/69585/travis_thumbnail.png 200w,\n/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png 387w",
                  sizes: "(max-width: 387px) 100vw, 387px",
                },
              },
            },
          },
        },
      },
    ],
  },
};
