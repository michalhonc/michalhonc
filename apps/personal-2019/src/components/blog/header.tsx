import * as React from "react";
import { useIntl, Link, changeLocale } from "gatsby-plugin-intl";

import { Logo } from "../logo";
import { Nav } from "../nav";

import "../../styles/blog.scss";

interface IProps {
  canChangeLocal?: boolean;
}

export const Header: React.FC<IProps> = ({ canChangeLocal = true }) => {
  const intl = useIntl();

  return (
    <div className="Blog-header">
      <Logo />

      <Nav canChangeLocal={canChangeLocal}>
        <li>
          <Link to="/">{intl.formatMessage({ id: "aboutMe" })}</Link>
        </li>
        <li>
          <Link to="/blog">{intl.formatMessage({ id: "blog" })}</Link>
        </li>
        <li>
          <Link to="/#kontakty">{intl.formatMessage({ id: "contacts" })}</Link>
        </li>
      </Nav>
    </div>
  );
};
