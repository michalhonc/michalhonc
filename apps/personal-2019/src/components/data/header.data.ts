export const HeaderQuery = {
  background: {
    childImageSharp: {
      fluid: {
        src: "/static/45779372c1991322e2cf8821eb9b133f/74fd5/header-background.jpg",
        aspectRatio: 1.3333333333333333,
        srcSet:
          "/static/45779372c1991322e2cf8821eb9b133f/e3903/header-background.jpg 480w,\n/static/45779372c1991322e2cf8821eb9b133f/b31d1/header-background.jpg 960w,\n/static/45779372c1991322e2cf8821eb9b133f/74fd5/header-background.jpg 1920w,\n/static/45779372c1991322e2cf8821eb9b133f/cb67e/header-background.jpg 2483w",
        sizes: "(max-width: 1920px) 100vw, 1920px",
      },
    },
  },
  foreground: {
    childImageSharp: {
      fluid: {
        src: "/static/96b4dfdae8e006ed6b85bfb9aadd902b/0e6e2/header-foreground.png",
        aspectRatio: 1.3333333333333333,
        srcSet:
          "/static/96b4dfdae8e006ed6b85bfb9aadd902b/b460a/header-foreground.png 480w,\n/static/96b4dfdae8e006ed6b85bfb9aadd902b/6e898/header-foreground.png 960w,\n/static/96b4dfdae8e006ed6b85bfb9aadd902b/0e6e2/header-foreground.png 1920w,\n/static/96b4dfdae8e006ed6b85bfb9aadd902b/4e88d/header-foreground.png 2880w,\n/static/96b4dfdae8e006ed6b85bfb9aadd902b/049dd/header-foreground.png 3309w",
        sizes: "(max-width: 1920px) 100vw, 1920px",
      },
    },
  },
};
