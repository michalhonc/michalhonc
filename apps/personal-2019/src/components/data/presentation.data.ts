export const PresentationQuery = {
  mockup: {
    childImageSharp: {
      fluid: {
        aspectRatio: 0.7092198581560284,
        src: "/static/ea55a42170bb63d820b1b81f9df1bf59/3891b/mockup.png",
        srcSet:
          "/static/ea55a42170bb63d820b1b81f9df1bf59/3ba9f/mockup.png 100w,\n/static/ea55a42170bb63d820b1b81f9df1bf59/8ac63/mockup.png 200w,\n/static/ea55a42170bb63d820b1b81f9df1bf59/3891b/mockup.png 400w,\n/static/ea55a42170bb63d820b1b81f9df1bf59/8c332/mockup.png 600w,\n/static/ea55a42170bb63d820b1b81f9df1bf59/bc8e0/mockup.png 800w,\n/static/ea55a42170bb63d820b1b81f9df1bf59/8d364/mockup.png 1500w",
        sizes: "(max-width: 400px) 100vw, 400px",
      },
    },
  },
};
