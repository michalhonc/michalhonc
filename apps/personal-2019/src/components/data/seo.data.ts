export const SeoQuery = {
  site: {
    siteMetadata: {
      author: "Michal Honc",
      description:
        "První konzultace ZDARMA! Vytvořím Vám rychlé a moderní webové stránky, se kterýma Vy nebo Váš produkt uspějete. Sjednejte si konzultaci zdarma během jedné minuty",
      title: "Michal Honc",
    },
  },
};
