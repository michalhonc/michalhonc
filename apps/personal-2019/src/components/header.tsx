import * as React from "react";
import { useStaticQuery, graphql } from "gatsby";
import BackgroundImage from "gatsby-background-image";
import { Link, useIntl, changeLocale } from "gatsby-plugin-intl";

import { Logo } from "./logo";
import Translate from "../images/translate.svg";
import { ReservationConsultation } from "./reservationConsultation";
import { Nav } from "./nav";

const Query = graphql`
  {
    background: file(relativePath: { eq: "header-background.jpg" }) {
      childImageSharp {
        fluid(maxWidth: 1920, quality: 90) {
          ...GatsbyImageSharpFluid_withWebp
        }
      }
    }
    foreground: file(relativePath: { eq: "header-foreground.png" }) {
      childImageSharp {
        fluid(maxWidth: 1920, quality: 100) {
          ...GatsbyImageSharpFluid_withWebp
        }
      }
    }
  }
`;

export const Header: React.FC = () => {
  const intl = useIntl();
  const data = useStaticQuery(Query);

  const { fluid: background } = data.background.childImageSharp;
  const { fluid: foreground } = data.foreground.childImageSharp;

  return (
    <header className="Header">
      <BackgroundImage
        Tag="div"
        className="Header-background"
        fluid={background}
      />
      <BackgroundImage
        Tag="div"
        className="Header-foreground"
        fluid={foreground}
      />

      <div className="Header-hero">
        <h1>
          <div>{intl.formatMessage({ id: "hero_1" })}</div>
          <div
            dangerouslySetInnerHTML={{
              __html: intl.formatMessage({ id: "hero_2" }),
            }}
          />
          <div>{intl.formatMessage({ id: "hero_3" })}</div>
          <div
            dangerouslySetInnerHTML={{
              __html: intl.formatMessage({ id: "hero_4" }),
            }}
          />
        </h1>
        <ReservationConsultation />
      </div>

      <Logo />

      <Nav canChangeLocal>
        <li>
          <Link to="/blog">{intl.formatMessage({ id: "blog" })}</Link>
        </li>
        <li>
          <a href="#kontakty">{intl.formatMessage({ id: "contacts" })}</a>
        </li>
      </Nav>

      {/* <button className="Header-button">
        <span><FormattedMessage id="download_CV" /></span>
      </button> */}
    </header>
  );
};
