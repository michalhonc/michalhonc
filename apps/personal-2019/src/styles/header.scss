$blur: 0.2rem;

@keyframes blur {
  0% {
    filter: blur(0);
  }
  100% {
    filter: blur($blur);
  }
}
@keyframes slideInFromRight {
  0% {
    transform: translateX(20rem);
  }
  100% {
    transform: translateX(0);
  }
}
@keyframes slideInFromLeft {
  0% {
    transform: translateX(-20rem);
  }
  100% {
    transform: translateX(0);
  }
}

.Header {
  position: relative;
  width: calc(100% + 2 * #{$blur});
  height: 100vh;
  min-height: 50rem;
  overflow: hidden;

  &-foreground,
  &-background {
    position: absolute;
    width: calc(100% + 2 * #{$blur});
    height: 101vh;

    @include respond('phone') {
      width: calc(100% + 15rem);
    }
  }

  &-background {
    animation: 0.5s ease-out 0s blur;
    animation-fill-mode: both;
    min-height: 50rem;
    top: -$blur - 0.2rem;
    left: -$blur;
    right: -$blur;
  }

  &-foreground {
    animation: 1s ease-out 0s slideInFromLeft;
    animation-fill-mode: both;
    margin-top: -100vh;
    top: 5rem;
  }

  &-navigation {
    position: absolute;
    top: 2rem;
    right: 15%;
    display: flex;
    text-transform: uppercase;

    @include respond('tab-port') {
      flex-direction: column;
      align-items: flex-end;
    }



    & li {
      &:not(:last-of-type) {
        margin-right: 3rem;
        @include respond('tab-port') {
          margin-right: 0;
          margin-bottom: 1rem;
        }
      }

      & a {
        color: #fff;
        font-size: 1.8rem;

        &:hover {
          color: #dedede;
        }
      }
    }

  }

  &-logo {
    height: 29.5rem;
    width: 40rem;
    position: absolute;
    top: 0;
    left: 5%;

    @include respond('phone') {
      left: -10%;
      width: 20rem;
      height: 19.5rem;
    }

    & > div {
      position: relative;
    }

    &-background {
      position: absolute;
      top: 0;

      @include respond('phone') {
        width: 20rem;
        height: 14.8rem;
      }
    }

    &-foreground {
      position: absolute;
      top: 3rem;
      left: 50%;
      transform: translateX(-50%);

      @include respond('phone') {
        top: 2rem;
        width: 11rem;
        height: 3.5rem;
      }

    }
  }

  &-changeLocale {
    position: absolute;
    height: 2.4rem;
    width: 2.4rem;
    top: 2rem;
    right: 5%;
  }

  &-hero {
    animation: 1s ease-out 0s slideInFromRight;
    animation-fill-mode: both;
    position: absolute;
    bottom: 40%;
    left: 50%;
    margin-right: 1.6rem;

    @include respond('phone') {
      left: 40%;
    }

    & .ReservationConsultation {
      margin-top: 5rem;

      @include respond('phone') {
        margin-top: 3rem;
      }
    }

    & h1 {
      & div {
        font-size: 2.4rem;
        color: #fff;

        @include respond('phone') {
          font-size: 1.4rem;
        }

        &:not(:last-of-type) {
          margin-bottom: 2rem;
          @include respond('phone') {
            margin-bottom: 0.5rem;
          }
        }

        &:nth-of-type(2) {
          font-size: 4rem;

          @include respond('phone') {
            font-size: 2.4rem;
          }
        }

        &:nth-of-type(4) {
          font-size: 3rem;

          @include respond('phone') {
            font-size: 1.8rem;
          }
        }
      }
    }
  }
}
