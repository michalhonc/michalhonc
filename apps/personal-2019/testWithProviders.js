import React from "react";
import renderer from "react-test-renderer";
import { IntlContextProvider } from "gatsby-plugin-intl";

export const renderWithProviders = (component) =>
  renderer.create(
    <IntlContextProvider value={{ language: "en-US", routed: true }}>
      {component}
    </IntlContextProvider>
  );
