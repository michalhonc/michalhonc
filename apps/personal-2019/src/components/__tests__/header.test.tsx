import * as React from "react";
import * as Gatsby from "gatsby";
import renderer from "react-test-renderer";
import { renderWithProviders } from "../../../testWithProviders";
import { Header } from "../header";
import { HeaderQuery } from "../data/header.data";

describe("header", () => {
  const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery");
  useStaticQuery.mockImplementation(() => HeaderQuery);

  it("renders correctly", () => {
    const tree = renderWithProviders(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
