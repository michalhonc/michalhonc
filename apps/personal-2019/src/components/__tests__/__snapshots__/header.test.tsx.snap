// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders correctly 1`] = `
<header
  className="Header"
>
  <div
    className="Header-background gbi--1076601012-fw4frLGxhPUKyvAJLnvQwh gatsby-image-wrapper"
    style={
      Object {
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "opacity": 0.99,
        "position": "relative",
      }
    }
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
          .Header-background.gbi--1076601012-fw4frLGxhPUKyvAJLnvQwh:before,
          .Header-background.gbi--1076601012-fw4frLGxhPUKyvAJLnvQwh:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            
            transition: opacity 0.5s ease 0.25s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .Header-background.gbi--1076601012-fw4frLGxhPUKyvAJLnvQwh:before {
            z-index: -100;
            
            
            opacity: 0; 
          }
          .Header-background.gbi--1076601012-fw4frLGxhPUKyvAJLnvQwh:after {
            z-index: -101;
            
            
            
          }
        ",
        }
      }
    />
    <noscript>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "
    .Header-background.gbi--1076601012-fw4frLGxhPUKyvAJLnvQwh:before {
      opacity: 1;
      background-image: url('/static/45779372c1991322e2cf8821eb9b133f/74fd5/header-background.jpg');
    }",
          }
        }
      />
    </noscript>
  </div>
  <div
    className="Header-foreground gbi-779337082-vHngSHeq1Tkwsc3cVa5vNN gatsby-image-wrapper"
    style={
      Object {
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "opacity": 0.99,
        "position": "relative",
      }
    }
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
          .Header-foreground.gbi-779337082-vHngSHeq1Tkwsc3cVa5vNN:before,
          .Header-foreground.gbi-779337082-vHngSHeq1Tkwsc3cVa5vNN:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            
            transition: opacity 0.5s ease 0.25s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .Header-foreground.gbi-779337082-vHngSHeq1Tkwsc3cVa5vNN:before {
            z-index: -100;
            
            
            opacity: 0; 
          }
          .Header-foreground.gbi-779337082-vHngSHeq1Tkwsc3cVa5vNN:after {
            z-index: -101;
            
            
            
          }
        ",
        }
      }
    />
    <noscript>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "
    .Header-foreground.gbi-779337082-vHngSHeq1Tkwsc3cVa5vNN:before {
      opacity: 1;
      background-image: url('/static/96b4dfdae8e006ed6b85bfb9aadd902b/0e6e2/header-foreground.png');
    }",
          }
        }
      />
    </noscript>
  </div>
  <div
    className="Header-hero"
  >
    <h1>
      <div>
        Jsem
      </div>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<b>VÝVOJÁŘ</b> a <b>NÁVRHÁŘ</b>",
          }
        }
      />
      <div>
        Který
      </div>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "vytváří <b>RYCHLÉ</b> a <b>MODERNÍ</b> webové stránky",
          }
        }
      />
    </h1>
    <button
      className="ReservationConsultation"
      onClick={[Function]}
      type="button"
    >
      Rezervovat konzultaci zdarma
    </button>
  </div>
  <div
    className="Header-logo"
  >
    <div>
      <test-file-stub
        className="Header-logo-background"
      />
      <a
        aria-label="Michal Honc Logo"
        href="/en-US/"
        onClick={[Function]}
      >
        <test-file-stub
          className="Header-logo-foreground"
        />
      </a>
    </div>
  </div>
  <nav>
    <ul
      className="Header-navigation"
    >
      <li>
        <a
          href="/en-US/blog"
          onClick={[Function]}
        >
          Blog
        </a>
      </li>
      <li>
        <a
          href="#kontakty"
        >
          Kontakty
        </a>
      </li>
    </ul>
    <button
      aria-label="Change language to english"
      className="Header-changeLocale"
      onClick={[Function]}
      type="button"
    >
      <test-file-stub />
    </button>
  </nav>
</header>
`;
